core/src/main/java/org/elasticsearch/common/lucene/search/function/FieldValueFactorFunction.java

```diff
@@ -54,13 +54,24 @@ public FieldValueFactorFunction(String field, float boostFactor, Modifier modifi
 
     @Override
     public LeafScoreFunction getLeafScoreFunction(LeafReaderContext ctx) {
-        final SortedNumericDoubleValues values = this.indexFieldData.load(ctx).getDoubleValues();
+        final SortedNumericDoubleValues values;
+        if(indexFieldData == null) {
+            values = null;
```

Comment:
can you do `values = FieldData. emptySortedNumericDoubles();` instead so that the LeafScoreFunction doesn't have to care about nulls?